src/NzbDrone.Core/Download/Clients/qBittorrent/QBittorrent.cs

```diff
@@ -0,0 +1,228 @@
+            }
+
+            var queueItems = new List<DownloadClientItem>();
+
+            foreach (var torrent in torrents)
+            {
+                var item = new DownloadClientItem();
+                item.DownloadId = torrent.Hash;
+                item.Title = torrent.Name;
+                item.TotalSize = torrent.Size;
+                item.DownloadClient = Definition.Name;
+                item.RemainingSize = (long)(torrent.Size * (1.0 - torrent.Progress));
+                if (torrent.Eta != -1)
+                    item.RemainingTime = TimeSpan.FromSeconds(torrent.Eta);
+
+                var outputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, status.OutputRootFolders[0]);
+
+                if (outputPath == null || outputPath.FileName == torrent.Name)
+                    item.OutputPath = outputPath;
+                else
+                    item.OutputPath = outputPath + torrent.Name;
+
+                switch (torrent.State)
+                {
+                    case "error": // some error occurred, applies to paused torrents
+                        item.Status = DownloadItemStatus.Warning;
+                        item.Message = "QBittorrent is reporting an error";
+                        break;
+
+                    case "pausedUP": // torrent is paused and has finished downloading
+                    case "pausedDL": // torrent is paused and has NOT finished downloading
+                        item.Status = DownloadItemStatus.Paused;
+                        break;
+
+                    case "queuedUP": // queuing is enabled and torrent is queued for upload
+                    case "queuedDL": // queuing is enabled and torrent is queued for download
+                    case "checkingUP": // torrent has finished downloading and is being checked
+                    case "checkingDL": // same as checkingUP, but torrent has NOT finished downloading
+                        item.Status = DownloadItemStatus.Queued;
+                        break;
+
+                    case "uploading": // torrent is being seeded and data is being transfered
+                    case "stalledUP": // torrent is being seeded, but no connection were made
+                        item.Status = DownloadItemStatus.Completed;
+                        break;
+
+                    case "downloading": // torrent is being downloaded and data is being transfered
+                        item.Status = DownloadItemStatus.Downloading;
+                        break;
+
+                    case "stalledDL": // torrent is being downloaded, but no connection were made
+                        item.Status = DownloadItemStatus.Warning;
+                        item.Message = "The download is stalled with no connections";
+                        break;
+
+                    default:
+                        throw new Exception("Invalid torrent state " + torrent.State);
```

Comment:
If a new state is added this will break the integration completely since each torrent isn't handled in a try/catch block. We either need to wrap each torrent processing in a try/catch and log the error or log an error here and default the state to something, downloading maybe? Its kind of a toss up which one is best, I'll let the other devs weigh in here.